### PR TITLE
Compare by fingerprint

### LINKED ISSUE #384: Compare keys by fingerprint
Later this should be compared by `primaryFingerprint` vs `$0.fingerprint` - it's ok to do it in a followup PR

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/pull/375#discussion_r666438234_

### PR DESCRIPTION
**Pull request by Kharchevskyi**
This PR adds functionality to compare keys by fingerprint

close #384

----------------------------------

**Tests**:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift [hunk 1]
```diff
@@ -90,17 +90,34 @@ final class MessageService {
             if isWrongPassPhraseError != nil {
                 reject(MessageServiceError.wrongPassPhrase(rawMimeData, passPhrase))
             } else {
-                keys
-                    .map { PassPhrase(value: passPhrase, longid: $0.longid) }
-                    .forEach { self.passPhraseService.savePassPhrase(with: $0, inStorage: false) }
+                // TODO: - Tom
```
Comment (Kharchevskyi) [inline FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift:93]: @tomholub can you please check?
I think related to #362

**Review by Kharchevskyi — COMMENTED** [review]
LGTM

File: FlowCrypt/Controllers/SignIn/SignInViewController.swift [hunk 2]
```diff
@@ -131,7 +131,8 @@ extension SignInViewController {
                 PrvKeyInfo(
                     private: TestData.k3rsa4096.prv,
                     longid: TestData.k3rsa4096.longid,
-                    passphrase: TestData.k3rsa4096.passphrase
+                    passphrase: TestData.k3rsa4096.passphrase,
+                    fingerprints: []
                 )
             ]
```
Comment (tomholub) [inline FlowCrypt/Controllers/SignIn/SignInViewController.swift:137]: Let's make another issue to remove this whole debugging code except for the `showToast`

**Review by tomholub — APPROVED** [review]
Thank you, this looks good. Didn't test functionality.

**Pull request merged**